src/utils/compat.ts

```diff
@@ -83,6 +86,33 @@ const decodeAddress = (asset: string, addr: string): DecodedAddress => {
     };
 };
 
+const probeAddress = (address: string): string | null => {
+    let asset: string | null = null;
+    btcPrefixes.forEach((prefix: string) => {
+        if (address.startsWith(prefix)) {
+            asset = BTC;
+        }
+    });
+    liquidPrefixes.forEach((prefix: string) => {
+        if (address.startsWith(prefix)) {
+            asset = LBTC;
+        }
+    });
+    if (asset !== null) {
+        return asset;
+    }
+    // fallback to more costly decoding
```

Comment:
Did you measure the performance of decoding the address?

Reply:
i did not measure it, but noticed the browser hanging sometimes decoding the address